packages/session-replay-browser/src/events/event-compressor.ts

```diff
@@ -0,0 +1,95 @@
+import type { eventWithTime } from '@amplitude/rrweb-types';
+import { SessionReplayJoinedConfig } from 'src/config/types';
+import { SessionReplayEventsManager } from 'src/typings/session-replay';
+import { pack } from '@amplitude/rrweb';
+import { getGlobalScope } from '@amplitude/analytics-client-common';
+
+interface TaskQueue {
+  event: eventWithTime;
+  sessionId: number;
+}
+
+export class EventCompressor {
+  taskQueue: TaskQueue[] = [];
+  isProcessing = false;
+  eventsManager?: SessionReplayEventsManager<'replay' | 'interaction', string>;
+  config: SessionReplayJoinedConfig;
+  deviceId: string | undefined;
+  canUseIdleCallback: boolean | undefined;
```

Comment:
I think the code could be cleaner if these instance variables were not possibly undefined. It seems like all are set in the constructor, so I don't think we need to recheck them in the class methods.